runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -152,9 +160,25 @@ public final class Engine implements AutoCloseable
         final Map<String, Guard> guardsByType = guards.stream()
             .collect(Collectors.toMap(g -> g.name(), g -> g));
 
-        final Callable<Void> configure =
-                new ConfigureTask(config.configURL(), schemaTypes, guardsByType::get, labels::supplyLabelId, maxWorkers, tuning,
-                        dispatchers, errorHandler, logger, context, config, extensions);
+        this.rootConfigURL = config.configURL();
+        if (this.rootConfigURL == null)
+        {
+            this.watcherTask = new FileWatcherTask(this::reconfigure);
+        }
+        else if ("http".equals(this.rootConfigURL.getProtocol()) || "https".equals(this.rootConfigURL.getProtocol()))
+        {
+            throw new UnsupportedOperationException();
+        }
+        else
+        {
+            this.watcherTask = new FileWatcherTask(this::reconfigure);
+        }
```

Comment:
Instead of assuming we only support http, https and file, then testing for the ones we don't support, better to test for the one we do support (file) and fail if not that, then continue without `else` as throw will escape out of the entire code block anyway.

Reply:
Done. Just a note here: we need to watch for "jar" as protocol, since in the tests, the configs are inside a jar.

Reply:
Is there any issue with `jar:` protocol config URLs?

Reply:
No problem with them. Just need to use FIleWatcher for them, so in the if statement, we need it as well.